tests/rpc-tests/rpc-test.js

```diff
@@ -0,0 +1,777 @@
+        console.log('📍 Executing: ', getTestData[i].req.method)
+        response = await axiosInstance.post('/', getTestData[i].req)
+
+        if (getTestData[i].req.method === 'bor_getCurrentProposer') {
+          const currentSnapshotResponse = await axiosInstance.post(
+            '/',
+            getCurrentSnapshot
+          )
+          // If the validator set changes, the assertion will fail
+          response = await axiosInstance.post('/', getTestData[i].req)
+          assert.deepStrictEqual(
+            currentSnapshotResponse.data.result.validatorSet.proposer.signer,
+            response.data.result
+          )
+          continue
+        }
+
+        if (getTestData[i].req.method === 'bor_getCurrentValidators') {
+          const currentSnapshotResponse = await axiosInstance.post(
+            '/',
+            getCurrentSnapshot
+          )
+          // If the validator set changes, the assertion will fail
+          response = await axiosInstance.post('/', getTestData[i].req)
+          assert.deepStrictEqual(
+            currentSnapshotResponse.data.result.validatorSet.validators,
+            response.data.result
+          )
+          continue
+        }
+
+        if (getTestData[i].req.method === 'eth_createAccessList') {
+          assert.deepStrictEqual(
+            sortAccessList(response.data.result.accessList),
+            sortAccessList(getTestData[i].res.result.accessList)
+          )
+          assert.deepStrictEqual(
+            response.data.result.gasUsed,
+            getTestData[i].res.result.gasUsed
+          )
+        } else {
+          assert.deepStrictEqual(response.data, getTestData[i].res)
+        }
+      }
+    }
+
+    // Sender rpc calls
+    const sendTestData = fetchTestFiles(
+      `../../tests/rpc-tests/RPC-testdata/${process.env.RPC_NETWORK}/senders`
+    )
+
+    let nonce = await web3.eth.getTransactionCount(sender)
+
+    const getFilterChanges = {
+      method: 'eth_getFilterChanges',
+      params: [''],
+      id: 1,
+      jsonrpc: '2.0'
+    }
+    let filterResponse
+
+    const getFilterLogs = {
+      method: 'eth_getFilterLogs',
+      params: [''],
+      id: 1,
+      jsonrpc: '2.0'
+    }
+    let filterLogsResponse
+
+    let senderrIterattions = process.env.EXECUTION_COUNT_SENDERS
+    if (
+      senderrIterattions <= 0 ||
+      senderrIterattions === null ||
+      senderrIterattions === undefined
+    ) {
+      senderrIterattions = 1
+    }
+
+    // Finally, fire 'em
+    for (let iter = 0; iter < senderrIterattions; iter++) {
+      await updateSenderTestData(
+        sendTestData,
+        web3,
+        sender,
+        nonce,
+        axiosInstance
+      )
+      console.log('📍Executing sender rpc calls. Iteration: ', iter)
+      for (let i = 0; i < sendTestData.length; i++) {
+        if (
+          process.env.RPC_URL &&
+          (sendTestData[i].req.method === 'eth_sendRawTransaction' ||
```

Comment:
whenever methods require additional data such as sensitive privateKey, we can introduce a test-data flag to skip them (or - in case like the filters - insert additional info in the request, where such data will be embedded in the test structure)